tasks/create_app_secrets.yml

```diff
@@ -0,0 +1,48 @@
+---
+# Create secrets for an app
+
+- name: Print app name
+  debug: msg="App name {{ app.name }}"
+  tags: secret
+
+# Decrypt vaulted credentials and register them in a credentials new
+# variable
+#
+# Nota bene: you will need to set the vaut password while using this
+# playbook to ensure we can decrypt vaulted credentials
+- name: Decrypt vaulted credentials
+  include_vars:
+    name: "{{ app.name }}"
+    file: "{{ item.path }}"
+  with_items: "{{ app | json_query(\"secrets[?customer=='\" + customer + \"' && environment=='\" + env_type + \"']\") }}"
+  register: credentials
+
+# For every credentials file, compile the OpenShift secret object template
+# (see apps/_common/secret/env_vars.yml.j2) given a group (the application,
+# _e.g._ richie or edxapp) and the set of environment variables defined in
+# the above mentionned credentials file.
+#
+# Nota bene: we store all secrets separated by a '||' in the project_secrets
+# variable, to iterate over secrets, you will need to split('||') that
+# variable first.
+- name: set_fact project_secrets
+  set_fact:
+    project_secrets: |
+      {% for item in credentials.results -%}
+        {% for group, env_vars in item.ansible_facts.iteritems() -%}
+          {% include "apps/_common/templates/secret/env_vars.yml.j2" -%}
+        {%- endfor %}
+        {% if not loop.last -%}||{% endif -%}
```

Comment:
I didn't really understand but it looks akward to || ?

Reply:
It was an ugly first implementation. This morning I came up with something way more clean a la Ansible :wink: